### PR TITLE
Retry creation of assignment topic a few times before giving up

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation
Currently, if assignment topic creation fails, we give up(unless it happens to be a timeout exception). With this change, we retry the creation a few times regardless of the exception before giving up. During startup, this gives more time for the rest of the system to be ready.
### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by jerrypeng** [comment]
@srkukarni I think we also need to do this for all the other producer/consumer creations in the worker service

File: pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java [hunk 1]
```diff
@@ -105,25 +105,12 @@ public SchedulerManager(WorkerConfig workerConfig, PulsarClient pulsarClient, Pu
                     .compressionType(CompressionType.LZ4)
                     .sendTimeout(0, TimeUnit.MILLISECONDS)
                     .createAsync().get(10, TimeUnit.SECONDS);
+            } catch (Exception e) {
```
Comment (sijie) [inline pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java]: I don't think we should retry at `InterruptedException` and also when `InterruptedException` is caught, you should propagate the interrupted bit to your outer thread `Thread.currentThread().interrupt()`

Comment (srkukarni) [reply pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java]: @sijie I've fixed the handling of the interrupted exception

**Comment by srkukarni** [comment]
run integration tests
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by jerrypeng** [comment]
rerun cpp tests
rerun integration tests
rerun java8 tests

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
1e344e0

**Pull request merged**